src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -93,8 +93,16 @@ private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
                 switch (containingMember)
                 {
                     case MethodDeclarationSyntax method:
+                        var returnType = method.ReturnType;
+                        var asyncModifier = method.Modifiers.FirstOrDefault(m => m.Kind() == SyntaxKind.AsyncKeyword);
+                        if (asyncModifier.Kind() != SyntaxKind.None)
+                        {
+                            // async Task<string> M() { return null; }
+                            return tryGetSingleTypeArgument(returnType);
+                        }
+
                         // string M() { return null; }
-                        return method.ReturnType;
+                        return returnType;
```

Comment:
what about if you're in a local function?

Reply:
Good catch. This was previously missing a warning from the compiler (to trigger on), but that was since fixed. Added